avm/src/vm/opcode.rs

```diff
@@ -2072,51 +2073,49 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
   io!("waitop", |args, hand_mem| {
     Box::pin(async move {
       let ms = hand_mem.read_fixed(args[0]) as u64;
-      delay_for(Duration::from_millis(ms)).await;
+      sleep(Duration::from_millis(ms)).await;
       hand_mem
     })
   });
   io!("httpget", |args, mut hand_mem| {
     Box::pin(async move {
-      let url = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0]));
-      let http_res = reqwest::get(&url).await;
-      let mut is_ok = true;
-      let result_str = if http_res.is_err() {
-        is_ok = false;
-        format!("{}", http_res.err().unwrap())
-      } else {
-        let body = http_res.ok().unwrap().text().await;
-        if body.is_err() {
-          is_ok = false;
-          format!("{}", body.err().unwrap())
-        } else {
-          body.unwrap()
-        }
+      let (res, ok) = match TryInto::<Uri>::try_into(HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0]))) {
+        Err(ee) => (format!("{}", ee), false),
+        Ok(url) => match Client::builder().build::<_, Body>(HttpsConnector::new()).get(url.clone()).await {
+          Err(ee) => (format!("{}", ee), false),
+          Ok(mut response) => match hyper::body::to_bytes(response.body_mut()).await {
+            Err(ee) => (format!("{}", ee), false),
+            Ok(body) => if let Ok(res) = String::from_utf8(body.to_vec()) {
+              (res, true)
+            } else {
+              (format!("could not parse"), false)
+            },
+          },
+        },
       };
-      let result = if is_ok { 1i64 } else { 0i64 };
       hand_mem.write_fractal(args[2], &Vec::new());
-      hand_mem.push_fixed(args[2], result);
-      hand_mem.push_fractal(args[2], &HandlerMemory::str_to_fractal(&result_str));
+      hand_mem.push_fixed(args[2], if ok { 1i64 } else { 0i64 });
+      hand_mem.push_fractal(args[2], &HandlerMemory::str_to_fractal(&res));
       hand_mem
     })
   });
   io!("httppost", |args, mut hand_mem| {
     Box::pin(async move {
       let url = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[0]));
       let payload = HandlerMemory::fractal_to_string(hand_mem.read_fractal(args[1]));
-      let client = reqwest::Client::new();
-      let http_res = client.post(&url).body(payload.clone()).send().await;
+      let client = Client::builder().build::<_, Body>(HttpsConnector::new());
+      let http_res = client.request(Request::post(&url).body(payload.clone().into()).unwrap()).await;
```

Comment:
I didn't catch this before, but this has a similar potential runtime crash with the `unwrap` here. Can we also guard this if it's a credible failure scenario? (Presumably again due to malformed URLs?)